ui/src/app/pages/workspace/workspace-wrapper.tsx

```diff
@@ -133,156 +134,157 @@ const MultiRegionWorkspaceNotification = () => {
+  const [showNewCtNotification, setShowNewCtNotification] = useState(false);
 
-    useEffect(() => {
-      const updateStores = async (namespace) => {
-        runtimeDiskStore.set({
+  useEffect(() => {
+    const updateStores = async (namespace) => {
+      runtimeDiskStore.set({
+        workspaceNamespace: namespace,
+        gcePersistentDisk: undefined,
+      });
+      runtimeStore.set({
+        workspaceNamespace: namespace,
+        runtime: undefined,
+        runtimeLoaded: false,
+      });
+      pollAborter.abort();
+      const newPollAborter = new AbortController();
+      setPollAborter(newPollAborter);
+
+      try {
+        await LeoRuntimeInitializer.initialize({
           workspaceNamespace: namespace,
-          gcePersistentDisk: undefined,
+          pollAbortSignal: newPollAborter.signal,
+          maxCreateCount: 0,
+          maxResumeCount: 0,
         });
-        runtimeStore.set({
-          workspaceNamespace: namespace,
-          runtime: undefined,
-          runtimeLoaded: false,
+      } catch {
+        // Ignore InitialRuntimeNotFoundError.
+        // Ignore ExceededActionCountError. This is thrown when the runtime doesn't exist, or
+        // isn't started. Both of these scenarios are expected, since we don't want to do any lazy
+        // initialization here.
+        // Also ignore LeoRuntimeInitializationAbortedError - this is expected when navigating
+        // away from a page during a poll.
+        // Ideally, we would handle or log errors except the ones listed above.
+      }
+    };
+    const getWorkspaceAndUpdateStores = async (namespace, id) => {
+      try {
+        // No destructuring because otherwise it shadows the workspace in props
+        const wsResponse = await workspacesApi().getWorkspace(namespace, id);
+        currentWorkspaceStore.next({
+          ...wsResponse.workspace,
+          accessLevel: wsResponse.accessLevel,
         });
-        pollAborter.abort();
-        const newPollAborter = new AbortController();
-        setPollAborter(newPollAborter);
-
-        try {
-          await LeoRuntimeInitializer.initialize({
-            workspaceNamespace: namespace,
-            pollAbortSignal: newPollAborter.signal,
-            maxCreateCount: 0,
-            maxResumeCount: 0,
-          });
-        } catch {
-          // Ignore InitialRuntimeNotFoundError.
-          // Ignore ExceededActionCountError. This is thrown when the runtime doesn't exist, or
-          // isn't started. Both of these scenarios are expected, since we don't want to do any lazy
-          // initialization here.
-          // Also ignore LeoRuntimeInitializationAbortedError - this is expected when navigating
-          // away from a page during a poll.
-          // Ideally, we would handle or log errors except the ones listed above.
+        updateStores(wsResponse.workspace.namespace);
+        setWorkspace(wsResponse.workspace);
```

Comment:
looks good - I think we also need a call on line 226-227 for the case where we populate it from `nextWs`